sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionProcessorCloseReason.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   The reason for closing an <see cref="IPartitionProcessor" />.
+    /// </summary>
+    ///
+    public enum PartitionProcessorCloseReason
+    {
+        /// <summary>A close request was fired by the user.</summary>
+        Shutdown,
+
+        /// <summary>The ownership of the associated partition was lost.</summary>
+        OwnershipLost,
+
+        /// <summary>A non-retryable exception was thrown by the Event Hub Client.</summary>
+        EventHubException,
+
+        /// <summary>An unknown circumstance forced the processor to close.</summary>
+        Unknown
```

Comment:
I'd recommend moving this to the first slot, so that any uninitialized value takes on `Unknown` rather than assuming `Shutdown`.    The zero member is always used for expressions like `default(PartitionProcessorCloseReason)`